src/core/tests/type_prop/matmul.cpp

```diff
@@ -552,6 +552,6 @@ TEST(type_prop, matmul_propagate_label_on_b_input_after_reshape) {
     const auto& output_shape = matmul->get_output_partial_shape(0);
     const auto labels = get_shape_labels(output_shape);
 
-    ASSERT_THAT(labels, ElementsAre(my_label, 0, 0));
+    ASSERT_THAT(labels, ElementsAre((int)my_label, 0, 0));
```

Comment:
This cast should not be here.
The `my_label` should have correct type.
Change from `size_t` -> `ov::label_t`